livedata/src/main/java/io/getstream/chat/android/livedata/entity/ChannelEntity.kt

```diff
@@ -105,9 +118,17 @@ data class ChannelEntity(var type: String, var channelId: String) {
         return c
     }
 
-    /** updates last message and lastmessagedate on this channel entity */
-    fun updateLastMessageDate(messageEntity: MessageEntity) {
-        lastMessageAt = max(lastMessageAt, messageEntity.createdAt ?: messageEntity.createdLocallyAt)
+    /** updates last message and lastMessageAt on this channel entity */
+    fun addMessage(messageEntity: MessageEntity) {
+        val createdAt = messageEntity.createdAt ?: messageEntity.createdLocallyAt
+        val messageEntityCreatedAt = checkNotNull(createdAt) { "created at cant be null, be sure to set message.createdAt" }
+
+        val updateNeeded = messageEntity.id == lastMessage?.id
+        val newLastMessage = lastMessageAt == null || messageEntityCreatedAt.after(messageEntityCreatedAt)
```

Comment:
I think this is the comparison that we're meant to have here? Not sure how we ended up with the current version of the line.

```suggestion
        val newLastMessage = lastMessageAt == null || messageEntityCreatedAt.after(lastMessageAt)
```

Reply:
Yes, it is the needed comparison

Reply:
Good spot